stdlib/http/client.pyi

```diff
@@ -103,6 +103,7 @@ class HTTPMessage(email.message.Message):
 def parse_headers(fp: io.BufferedIOBase, _class: Callable[[], email.message.Message] = ...) -> HTTPMessage: ...
 
 class HTTPResponse(io.BufferedIOBase, BinaryIO):  # type: ignore[misc]  # incompatible method definitions in the base classes
+    url: str
```

Comment:
So, if I understand correctly, this attribute isn't set in the `__init__` method of this class, nor in any other method of this class; it's instead monkey-patched onto instances of this class by functions that live in `urllib.request` and/or `urllib.response`. Is that correct? If so, can we make it so that this is the last attribute annotation in the class body, and add a comment to that effect?

Reply:
Thanks for the response!  Just playing directly with `HTTPResponse`, I can see you're right.

```python
from http.client import HTTPResponse

response = HTTPResponse(some_sock)
response.url  # AttributeError
```

Note in the PR description there's justification for why the top line, though I now agree the bottom line makes more sense too (since `url` is not originally part of `HTTPResponse`, which is created by `urllib.request.HTTPSHandler`).

> it's instead monkey-patched onto instances of this class

This line in `urllib.request.AbstractHTTPHandler` adds the `url` attribute to the `HTTPResponse`: https://github.com/python/cpython/blob/main/Lib/urllib/request.py#L1334

What makes you say it's monkey patched?  To me, it seems like it's assigned.

---

Anyways, what do you think of the below comment?

```python
# NOTE: url is assigned by HTTP handler during do_open to match
# urllib.response.addinfourl's interface, not originally in HTTPResponse
url: str
```

Reply:
> What makes you say it's monkey patched? To me, it seems like it's assigned.

You say tomato, I say tomato :P

"Monkey patching" is always just assignment in Python, right? Here I just used the term to indicate that an attribute that was never assigned to inside the class was being assigned to outside the class.

> Anyways, what do you think of the below comment?
> 
> ```python
> # NOTE: url is assigned by HTTP handler during do_open to match
> # urllib.response.addinfourl's interface, not originally in HTTPResponse
> url: str
> ```

How about:

```python
# url is set on instances of the class in urllib.request.AbstractHTTPHandler.do_open
# to match urllib.response.addinfourl's interface.
# It's not set in HTTPResponse.__init__ or any other method on the class
url: str
```

Reply:
Oh I actually had a misunderstanding of the term monkey patching.  I thought it only applied when using `patch` from `unittest.mock`, now I understand it's really a general term for dynamic attribute assignment.

Regardless, updated the comment to match this, thank you!